pkg/quota/generic/evaluator.go

```diff
@@ -41,6 +41,13 @@ func ListResourceUsingInformerFunc(f informers.SharedInformerFactory, resource s
 		if err != nil {
 			return nil, err
 		}
+
+		// when the informer has not synced, should return error
+		// otherwise, the quota will be exceeded.
+		if !informer.Informer().HasSynced() {
+			return nil, fmt.Errorf("%s hasn't been synced", resource.String())
```

Comment:
Is there some way to call `controller.WaitForCacheSync`? just like what we have done in other controller?

Honestly, I'm not sure what'll happen if we just return an error? will it keep retry until synced?